src/Common/ShowResults.pas

```diff
@@ -2928,53 +2887,31 @@ procedure ShowIsolated(DSS: TDSSContext; FileNm: String);
                 if TestElement.Enabled then
                     if not (Flg.Checked in TestElement.Flags) then
                     begin
-                        FSWrite(F, '"' + TestElement.FullName + '"');
-                        FSWrite(F, '  Buses:');
+                        isolatedElement.name := TestElement.FullName;
                         for j := 1 to TestElement.nterms do
-                            FSWrite(F, '  "', TestElement.GetBus(j), '"');
-                        FSWriteln(F);
-                    end;
-                TestElement := CktElements.Next;
-            end;
-        end;
-
-        FSWriteln(F);
-        FSWriteln(F, '***  THE FOLLOWING BUSES ARE NOT CONNECTED TO ANY POWER DELIVERY ELEMENT ***');
-        FSWriteln(F);
-
-        with DSS.ActiveCircuit do
-        begin
-            for j := 1 to NumBuses do
-                if not Buses^[j].BusChecked then
-                    FSWriteln(F, EncloseQuotes(BusList.NameOfIndex(j)));
-        end;
-
+                        begin
+                            i := Length(isolatedElement.buses) + 1;
+                            SetLength(isolatedElement.buses, i); 
+                            isolatedElement.buses[i-1] := TestElement.GetBus(j);
+                        end;
 
-        FSWriteln(F);
-        FSWriteln(F, '***********  CONNECTED CIRCUIT ELEMENT TREE ************');
-        FSWriteln(F);
-        FSWriteln(F, '(Lexical Level) Element name');
-        FSWriteln(F);
+                        i := Length(isolatedBusesReport.isolatedElements) + 1;
+                        SetLength(isolatedBusesReport.isolatedElements, i); 
 
-        TestBranch := Branch_List.First;
-        while TestBranch <> NIL do
-        begin
-            FSWriteln(F, ['(', IntToStr(Branch_List.Level), ') ', TestBranch.ParentClass.Name, '.', TestBranch.Name]);
-            TestElement := Branch_List.FirstObject;
-            while TestElement <> NIL do
-            begin
-                FSWriteln(F, ['[SHUNT], ', TestElement.ParentClass.Name, '.', TestElement.Name]);
-                TestElement := Branch_List.NextObject
+                        isolatedElement.numBuses := Length(isolatedElement.buses);
+                        isolatedBusesReport.isolatedElements[i-1] := isolatedElement;
+                    end;
+                TestElement := CktElements.Next;
```

Comment:
The connected circuit element tree was dropped deliberately as we have no interest in it.

The dropped list of buses seems to me an absolute copy of the first list, so I dropped it as well.

Reply:
My reading indicates the second list should only include things with no source of energy, rather than those that are disconnected from the primary source. I would think this means if there is secondary generation in disconnected parts of the model they would be excluded from the second list.

Probably doesn't make a huge difference for our use case as we are including only network that is traced from the feeder head, so it should all be connected anyway

Reply:
I've looked at how these loops are structured and if something could affect it enough so that the second run would generate a different list - and it doesn't seem like it, so at this point I would just leave it out - if we notice an issue, I can restore it into probably a separate list.

Reply:
not a second run, a secondary source of energy. I would expect this to be something like a energy consumer with negative flow, or whatever you use to represent battery/solar/wind etc